### DISCUSSION TITLE
macOS Monterey — command line tools — brew and Xcode disagree

### DISCUSSION THREAD

**Discussion by cwreynolds** [Homebrew/discussions#3345] [Getting started] [answered]
I have a new M1 laptop with macOS Monterey 12.4 and Xcode 13.4.1.

Tried using Homebrew to install OpenCV. It said “Error: gcc: the bottle needs the Apple Command Line Tools to be installed.” 

But when I tried to run `xcode-select --install` it said “xcode-select: error: command line tools are already installed, use "Software Update" to install updates” I had installed the latest Xcode earlier today. The “Locations” in Preferences shows the version corresponding to my Xcode (13.4.1).

**Comment by carlocab** [comment]
What is the output of `brew config` and `brew doctor`?

**Reply by cwreynolds** [reply]
Thanks for your quick response!

```
> brew doctor
Your system is ready to brew.
> brew config
HOMEBREW_VERSION: 3.5.1
ORIGIN: https://github.com/Homebrew/brew
HEAD: 2258ba5797c1ea8149a49673b5c080011c366237
Last commit: 2 days ago
Core tap ORIGIN: https://github.com/Homebrew/homebrew-core
Core tap HEAD: 1d9fb5dd1e6ead16435e8bf8ec51fe3b5ec1d92f
Core tap last commit: 84 minutes ago
Core tap branch: master
HOMEBREW_PREFIX: /opt/homebrew
HOMEBREW_CASK_OPTS: []
HOMEBREW_CORE_GIT_REMOTE: https://github.com/Homebrew/homebrew-core
HOMEBREW_DISPLAY: /private/tmp/com.apple.launchd.LyIQSZXXuA/org.macosforge.xquartz:0
HOMEBREW_MAKE_JOBS: 10
Homebrew Ruby: 2.6.8 => /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/bin/ruby
CPU: 10-core 64-bit arm_firestorm_icestorm
Clang: 13.1.6 build 1316
Git: 2.32.1 => /Applications/Xcode.app/Contents/Developer/usr/bin/git
Curl: 7.79.1 => /usr/bin/curl
macOS: 12.4-arm64
CLT: N/A
Xcode: 13.4.1
Rosetta 2: false
```

**Comment by Bo98** [comment, marked as answer]
The CLT installed check works by checking if it can extract a valid version number either via `pkgutil` (which doesn't always work, but is still preferred) or falling back to invoking `/Library/Developer/CommandLineTools/usr/bin/clang --version` (which should always work). What does that clang command output for you?

One possibility is that the CLT is partially installed. You can reinstall by removing `/Library/Developer/CommandLineTools` and triggering another install. Just be aware that `xcode-select --install` is prone to installing old versions so check Software Update afterwards (you only need to select the latest if it prompts for multiple versions), or just download the latest CLT from developer.apple.com instead.

**Reply by toddjonker** [reply]
I'm having similar issues as reported at https://github.com/amazon-ion/homebrew-ion-cli/issues/2  In my case, brew says my CLI tools are "too outdated".  I don't see how I could have partially installed the tools though.

I think the error message could use more context, like "you have version X but we require at least Y", and to be clear that the message is from Homebrew itself, and not the tool I'm trying to install.

**Comment by cwreynolds** [comment]
Thanks. Your comments are super helpful. It appears that my CommandLineTools were “from the Truman administration” and not been correctly updated when I installed the new Xcode. I noticed in passing that it had not asked me the normal question about installing CommandLineTools, but I was willing to assume that was just one of the many things that had changed since the last time I did this:
```
> /Library/Developer/CommandLineTools/usr/bin/clang --version
zsh: bad CPU type in executable: /Library/Developer/CommandLineTools/usr/bin/clang

> ls -al /Library/Developer/CommandLineTools/          
total 0
drwxrwxr-x  5 root  wheel  160 Jun  7 08:50 .
drwxrwxr-x  5 root  wheel  160 Jun  7 08:37 ..
drwxr-xr-x  6 root  wheel  192 Nov 18  2019 Library
drwxr-xr-x  5 root  wheel  160 Jan 14  2020 SDKs
drwxr-xr-x  7 root  wheel  224 Dec 27  2019 usr

sudo rm -rf /Library/Developer/CommandLineTools

> xcode-select --install
xcode-select: note: install requested for command line developer tools
[... popup ... “The software was installed”]

> /Library/Developer/CommandLineTools/usr/bin/clang --version
Apple clang version 13.0.0 (clang-1300.0.27.3)
Target: arm64-apple-darwin21.5.0
Thread model: posix
InstalledDir: /Library/Developer/CommandLineTools/usr/bin
```
When I open App Store [Updates] it now says that Xcode was installed today. After this I was able to run `brew install opencv` normally.
```
> brew install opencv
...
==> Installing opencv
==> Pouring opencv--4.5.5_2.arm64_monterey.bottle.tar.gz
🍺  /opt/homebrew/Cellar/opencv/4.5.5_2: 854 files, 119.7MB
```
Thanks so much!

For what it's worth, I will file a bug report via Apple's Feedback Assistant, referring back to this ticket, just so they know that the upgrade (to a new laptop, new architecture, new OS) did not work correctly.

**Reply by cwreynolds** [reply]
Oh, less than an hour later I got a notification from Software Update (which I had been confusing with App Store):

    Updates are available for your Mac
    Update Now
    • Command Line Tools for Xcode 13.2
    • Command Line Tools for Xcode 13.3
    • Command Line Tools for Xcode 13.4

I installed those updates and all still SEEMS well. I assume I am using the 13.1 version.
